src/main/java/org/robotframework/remoteswinglibrary/agent/ServerThread.java

```diff
@@ -57,12 +53,11 @@ public void run()  {
         }
     }
 
-    private static void notifyPort(final Integer portToNotify, final String serverHost, final Integer serverPort) throws IOException {
-        Socket echoSocket = new Socket(serverHost, serverPort);
-        PrintWriter outToServer = new PrintWriter(echoSocket.getOutputStream(), true);
-        outToServer.write(portToNotify.toString() + ":" + getName());
-        outToServer.close();
-        echoSocket.close();
+    private void notifyPort(final Integer portToNotify) throws IOException {
+        robotConnection.connect();
+        //robotConnection.send("DIALOG:" + "aaa");
```

Comment:
uncomment this line to see problem with sending multiple messages with socket connection.